metadata/src/main/java/org/apache/kafka/controller/QuorumController.java

```diff
@@ -783,6 +798,11 @@ public void handleLeaderChange(LeaderAndEpoch newLeader) {
                 });
             } else if (curClaimEpoch != -1) {
                 appendControlEvent("handleRenounce[" + curClaimEpoch + "]", () -> {
+                    if (this != metaLogListener) {
```

Comment:
Maybe we can add a little helper? The only thing difference in all of these checks is the type of event.

Reply:
Good idea. Fixed.